backend/services/users.js

```diff
@@ -36,7 +36,10 @@ async function createUser(raw_user) {
   };
 
   delete raw_user_no_secret.secret;
-  return new User(raw_user_no_secret).save();
+  user = new User(raw_user_no_secret);
+  await user.save();
+  user.role = pending_role;
```

Comment:
Why is this line need if we have on line 35: `role: pending_role._id`

Reply:
This seems like a merge conflict that was improperly resolved. The line being deleted in that diff came from [this PR](https://github.com/TritonSE/TSE-Oktavian/commit/6d2e7929f500044ef893378eb1184330ec46f17b#diff-c8feca3408825c1b0293f25ad5e0a50288093429959e8ec7448e8a5f3837e9bcR39) where I simplified the logic.

Reply:
@AnsonLee99, could you please undo this change before merging?

Reply:
No, this code is needed because the role field needs to be populated for PageContainer to not throw an error after the registration process.. I fixed this in a previous PR but there was a recent commit that broke it again.